vscode-trace-extension/src/trace-explorer/trace-utils.ts

```diff
@@ -62,93 +63,101 @@ export const fileHandler =
+                                'Failed to open trace: ' +
+                                    traceName +
+                                    '. There may be an issue with the server or the trace is invalid.'
+                            );
+                        }
+                    }
 
-                progress.report({ message: ProgressMessages.MERGING_TRACES, increment: 40 });
-                if (traces === undefined || traces.length === 0) {
-                    progress.report({ message: ProgressMessages.COMPLETE, increment: 30 });
-                    return;
-                }
+                    if (token.isCancellationRequested) {
+                        rollbackTraces(traces, 20, progress);
+                        progress.report({ message: ProgressMessages.COMPLETE, increment: 50 });
+                        return;
+                    }
 
-                const experiment = await experimentManager.openExperiment(name, traces);
-                const panel = TraceViewerPanel.createOrShow(context.extensionUri, experiment?.name ?? name, undefined);
-                if (experiment) {
-                    panel.setExperiment(experiment);
-                }
+                    progress.report({ message: ProgressMessages.MERGING_TRACES, increment: 40 });
+                    if (traces === undefined || traces.length === 0) {
+                        progress.report({ message: ProgressMessages.COMPLETE, increment: 30 });
+                        return;
+                    }
 
-                if (token.isCancellationRequested) {
+                    const experiment = await experimentManager.openExperiment(name, traces);
+                    const panel = TraceViewerPanel.createOrShow(
+                        context.extensionUri,
+                        experiment?.name ?? name,
+                        undefined
+                    );
                     if (experiment) {
-                        experimentManager.deleteExperiment(experiment.UUID);
+                        panel.setExperiment(experiment);
+                    }
+
+                    if (token.isCancellationRequested) {
+                        if (experiment) {
+                            experimentManager.deleteExperiment(experiment.UUID);
+                        }
+                        rollbackTraces(traces, 20, progress);
+                        progress.report({ message: ProgressMessages.COMPLETE, increment: 10 });
+                        panel.dispose();
+                        return;
                     }
-                    rollbackTraces(traces, 20, progress);
-                    progress.report({ message: ProgressMessages.COMPLETE, increment: 10 });
-                    panel.dispose();
-                    return;
+                    progress.report({ message: ProgressMessages.COMPLETE, increment: 30 });
+                } finally {
+                    updateNoExperimentsContext();
```

Comment:
That's a creative way of ensuring to update that context after that "leaky" block executes, without adding it in a bunch of places - I like